pkg/conversion/capi2mapi/machine.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package capi2mapi
+
+import (
+	"strings"
+
+	mapiv1 "github.com/openshift/api/machine/v1beta1"
+	conversionutil "github.com/openshift/cluster-capi-operator/pkg/conversion/util"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+const (
+	mapiNamespace            = "openshift-machine-api"
+	workerUserDataSecretName = "worker-user-data"
+)
+
+// fromCAPIMachineToMAPIMachine translates a core CAPI Machine to its MAPI Machine correspondent.
+//
+//nolint:funlen
+func fromCAPIMachineToMAPIMachine(capiMachine *capiv1.Machine) (*mapiv1.Machine, error) {
+	errs := []*field.Error{}
+
+	mapiMachine := &mapiv1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        capiMachine.Name,
+			Namespace:   mapiNamespace,
+			Labels:      capiMachine.Labels,
+			Annotations: capiMachine.Annotations,
+			// OwnerReferences: These need to be converted so that any MachineSet owning a Machine is represented with the correct owner reference between the two APIs.
```

Comment:
OCPCLOUD-2716?

Reply:
I have it on line 63 already, but can add here as well if you prefer